### PR TITLE
MB-16790 Regenerate service items when address change updates pricing

### PR DESCRIPTION
**Pull request by kctruss**
## Summary
When an approved address change results in a conversion between shorthaul and linehaul, the system will reject previous service items and generate new ones. 

**Is there anything you would like reviewers to give additional scrutiny?**

- Let me know if there is a better way to clone a service item struct

### Setup to Run the Code

- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the office app as a prime user
2. Find a move and update the destination address to one that will change the move from linehaul to shorthaul and/or vice versa.
3. Approve the address change as a TOO
4. Confirm that the service items have been updated and that DLH was converted to DSH or vice versa

### CODE REVIEW

File: pkg/services/shipment_address_update/shipment_address_update_requester_test.go [hunk 1]
```diff
@@ -703,3 +827,15 @@ func (suite *ShipmentAddressUpdateServiceSuite) getServiceItemsByStatus(items mo
...
+	itemsWithCode := models.MTOServiceItems{}
+
+	for _, si := range items {
+		if si.ReService.Code == code {
+			itemsWithCode = append(itemsWithCode, si)
+		}
+	}
+
+	return itemsWithCode
+}
```
Comment (ruizajtruss) [inline pkg/services/shipment_address_update/shipment_address_update_requester_test.go:841]: This is great! 🎉

File: pkg/services/shipment_address_update/shipment_address_update_requester.go [hunk 2]
```diff
@@ -263,16 +310,27 @@ func (f *shipmentAddressUpdateRequester) ReviewShipmentAddressChange(appCtx appc
...
+				// Ensure that the updated pricing is applied (e.g. DLH -> DSH, DSH -> DLH etc.)
+				regeneratedServiceItem, _, createErr := serviceItemCreator.CreateMTOServiceItem(appCtx, &copyOfServiceItem)
+				if createErr != nil {
+					return nil, createErr
+				}
+				regeneratedServiceItems = append(regeneratedServiceItems, *regeneratedServiceItem...)
 			}
+
+			// Append the auto-generated service items to the shipment service items slice
+			addressUpdate.Shipment.MTOServiceItems = append(addressUpdate.Shipment.MTOServiceItems, regeneratedServiceItems...)
```
Comment (ruizajtruss) [inline pkg/services/shipment_address_update/shipment_address_update_requester.go:333]: **Non-blocking suggestion**: I just thought of this, should we encapsulate this in an `if` statement to check that the service item doesn't already have a rejected status?

**Review by ruizajtruss — APPROVED** [review]
Code looks great and everything passes! 🚀 Just left one small suggestion I thought about

File: pkg/services/shipment_address_update/shipment_address_update_requester.go (hunk 2 above)
Comment (kctruss) [reply pkg/services/shipment_address_update/shipment_address_update_requester.go:333]: Ohhh good call. I can see that ending up with some wonky data. I'll work on adding that check and a couple more tests.

**Pull request merged**